### PR TITLE
Addding first commit for useIntersectionObserver

### PR DESCRIPTION
**Pull request by simbathesailor**
Adding the commit for useIntersectionObserver. We can take the discussion from here. 

As discussed here [https://github.com/imbhargav5/rooks/issues/22#issuecomment-498124498](url)

I will be working on useLazyImage which may be using useIntersectionObserver indirectly.

Codesandbox link is:  https://codesandbox.io/s/useintersectionobserver-sbz54

### CODE REVIEW

**Comment by simbathesailor** [comment]
I am having hard time resoving conflicts for 
packages/shared/tsconfig.tsbuildinfo and yarn.lock. 

What should I do to resolve the conflicts efficiently?. 

What strategy people use generally for such kind of files?

For now those file , I deleted from pr. are they neccessary. I think that is generated everytime you build the project.

**Comment by imbhargav5** [comment]
Hey @simbathesailor I added you as collaborator. Can you work on this PR branch?

**Comment by imbhargav5** [comment]
Hey, if the two files are tsconfig.buildinfo and yarn.lock, they are autogenerated so if you have too many merge conflicts, resolve them as the ones you generated. It's fine.

**Comment by imbhargav5** [comment]
This looks very good. Also, let's add a storybook entry for `useIntersectionObserver` and a few tests.

**Comment by simbathesailor** [comment]
@imbhargav5 What should be the approach for testing some code which is using 
interSectionObserver. Do you have any pointers around how to approach this? 

I came nearest to mocking the intersectionObserver. But simulating everything in test cases will bring the test cases away from the actual implementation of useIntersectionObserver. Can you suggest something?

**Comment by simbathesailor** [comment]
@imbhargav5 
Added storybook entry with a basic example for now and updated README.md. I am still figuring out how to write test cases for such a feature.

**Comment by simbathesailor** [comment]
I have been busy lately. I will try to finalize this useIntersectionObserver hook in next week. I hope will be having time :)

**Comment by imbhargav5** [comment]
@simbathesailor No problem man.

File: packages/storybook/src/index.js [hunk 1]
```diff
@@ -26,3 +26,4 @@ import "./visibility-sensor";
 import "./will-unmount";
 import "./window-size";
 import "./worker";
+import "./intersectionobserver"
```
Comment (imbhargav5) [inline packages/storybook/src/index.js]: Let's put this in alphabetic order.

**Comment by simbathesailor** [comment]
WIll be rasiing the separate pr for it. The existing one is from the forked repo. I will be raising a pr from the rooks repo itself

**Pull request closed without merging**